ros/processor/inventory_events_consumer.py

```diff
@@ -134,6 +134,8 @@ def process_system_details(self, msg):
         """ Store new system information (stale, stale_warning timestamp) and return internal DB id"""
         host = msg['host']
         performance_record = get_performance_profile(msg['platform_metadata']['url'], host['account'])
+        region = performance_record.get('region')
+        del performance_record["region"]
```

Comment:
We are getting region from inside the `rule_hit_details` like this:
```python
{
	'rhel': '8.4',
	'cloud_provider': 'Amazon Web Services',
	'instance_type': 't2.micro',
	'region': 'us-east-1',
}
```
Then can we NOT choose this method to fetch it from insights-core, add it to the performance_record object, store in a variable and then delete that key again from the record object. Please let me know if we have done this way for any specific reason.
In case not, then I would vote for directly using `region` value from `reports` and similar for the `instance_type`. Thoughts?

Reply:
For reference -> As per https://github.com/RedHatInsights/ros-backend/pull/156, this was supposed to be done for handling Optimized state that has no reports and Waiting for data state that shows no region in the rule hit.